### DISCUSSION TITLE
Airflow Tasks failing without any logs in UI.

### DISCUSSION THREAD

**Discussion by Vishal2696** [apache/airflow#29438] [Q&A]
**Airflow Version:** 2.2.3
**Deployment:** Kubernetes (Azure AKS)
**Executor**: Celery
**Metadata Db:** Azure Postgres Single Server
**Using Redis:** Yes
**Connection to DB:** Via pgbouncer (running as a separate pod)

The issue I'm facing occurs randomly and not all the time. Currently not able identify any pattern. I have a airflow task that uses `ExternalTaskSensor` to wait for another dag. The task runs fine most of the time until it fails randomly. 

In the web UI below is the logs I see for the failed task. 
```
*** Log file does not exist: /opt/airflow/airflow/logs/mydagname/mytaskname/2023-02-05T04:00:00+00:00/9.log
*** Fetching from: http://worker:8793/log/mydagname/mytaskname/2023-02-05T04:00:00+00:00/9.log
*** Failed to fetch log file from worker. 404 Client Error: NOT FOUND for url: http://worker:8793/log/mydagname/mytaskname/2023-02-05T04:00:00+00:00/9.log
For more information check: https://httpstatuses.com/404
```

Below is the logs I see from the worker pod during the same time. 
```
ERROR/ForkPoolWorker-7] Task airflow.executors.celery_executor.execute_command[40ed60de-76ee-4694-b8af-e23e69a5e7e8] raised unexpected: AirflowException('Celery command failed on host: worker')
raise AirflowException('Celery command failed on host: ' + get_hostname())
airflow.exceptions.AirflowException: Celery command failed on host: worker
```

Below is the logs I see from the scheduler pod during the same time. 
```
ERROR[0m - Executor reports task instance <TaskInstance: mydagname.mytaskname scheduled__2023-02-04T08:30:00+00:00 [queued]> finished (failed) although the task says its queued. (Info: Celery command failed on host: worker) Was the task killed externally?
```
I've checked the redis and pgbouncer logs and they don't report any errors. The connection to DB also doesn't seem to have any failures. The memory and CPU usage also seem to be stable. Any idea on what could cause this? Which area should I be looking at?

**Comment by potiuk** [comment]
There might be a number of reasons https://github.com/apache/airflow/issues/16025 - and you can look for some remedies. I think the last one (not enough disk space for log?) might be an interesting clue. Also this one is a possible reason https://github.com/apache/airflow/issues/13824 . Usually I think this is something preventing the file from being written.  The bad thing is that there is ... no log file to tell you what's wrong. So this is a bit chicken-egg problem when you look at this from Airflow perspective. But I think maybe you should take a look at flower and see if there are failing celery tasks and more information will be shown there in celery internals. Possibly also adding debugging for celery could help ?
